### DISCUSSION TITLE
Survey: What IDE do you use in your Mill projects?

### DISCUSSION THREAD

**Discussion by lefou** [com-lihaoyi/mill#3341] [General]
* Use one IDE/editor per main-comment
* Use the 👍‍ to add/count yourself
* Use sub-comment is you need to explain

**Comment by lefou** [comment]
IntelliJ IDEA

**Reply by lefou** [reply]
with `mill.idea.GenIdea`

**Reply by lefou** [reply]
with `mill.bsp.BSP` (Mill built-in BSP server, `mill --bsp`)

**Reply by lihaoyi** [reply]
I use the IntelliJ BSP import button. I guess that's mill.bs.BSP?

**Reply by lefou** [reply]
with `mill.contrib.bloop.Bloop`

**Comment by lefou** [comment]
Metals + VSCode

**Reply by lefou** [reply]
with `mill.bsp.BSP` (Mill built-in BSP server, `mill --bsp`)

**Reply by lefou** [reply]
with `mill.contrib.bloop.Bloop`

**Reply by kubukoz** [reply]
contrib, because that's what Metals defaults to :)

**Comment by lefou** [comment]
Eclipse

**Comment by kdoomsday** [comment]
Metals + emacs

**Comment by ayhon** [comment]
Metals + neovim
